### PR TITLE
Add flags to avoid recording calls in the database and caching.

### PR DESCRIPTION
**Pull request by schmmd**

### CODE REVIEW

**Comment by schmmd** [comment]
@joelgrus ping

File: allennlp/service/server_flask.py [hunk 1]
```diff
@@ -154,6 +154,12 @@ def predict(model_name: str) -> Response:  # pylint: disable=unused-variable
         if request.method == "OPTIONS":
             return Response(response="", status=200)
 
+        # Do log if no argument is specified
+        record_flag = not request.args.get("record", "true").lower() == "false"
```
Comment (joelgrus) [inline allennlp/service/server_flask.py]: `!=` would be more idiomatic (and more readable) than `not ==`

File: allennlp/service/server_flask.py [hunk 2]
```diff
@@ -154,6 +154,12 @@ def predict(model_name: str) -> Response:  # pylint: disable=unused-variable
...
+
+        # Do use the cache if no argument is specified
+        cache_flag = not request.args.get("cache", "true").lower() == "false"
```
Comment (joelgrus) [inline allennlp/service/server_flask.py]: same comment

File: allennlp/service/server_flask.py (hunk 1 above)
Comment (joelgrus) [reply allennlp/service/server_flask.py]: also, I would prefer more descriptive names than `_flag`, e.g. `record_to_database` and `check_cache`

File: allennlp/service/server_flask.py [hunk 3]
```diff
@@ -162,39 +168,44 @@ def predict(model_name: str) -> Response:  # pylint: disable=unused-variable
 
         log_blob = {"model": model_name, "inputs": data, "cached": False, "outputs": {}}
 
-        # See if we hit or not. In theory this could result in false positives.
-        pre_hits = _caching_prediction.cache_info().hits  # pylint: disable=no-value-for-parameter
+        if cache_flag:
```
Comment (joelgrus) [inline allennlp/service/server_flask.py]: I feel like you made this too complicated, I would have just changed line 169 to

```
if check_cache and cache_size > 0
```

and then line 182 to

```
if record_to_database and demo_db is not None:
```

is there a reason why that wouldn't just work?

**Comment by schmmd** [comment]
Thanks @joelgrus, all great feedback!

**Pull request merged**